system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs

```diff
@@ -1889,6 +1961,173 @@ impl_runtime_apis! {
 		}
 	}
 
+	impl pallet_revive::ReviveApi<Block, AccountId, Balance, Nonce, BlockNumber> for Runtime
+	{
+		fn balance(address: H160) -> U256 {
+			Revive::evm_balance(&address)
+		}
+
+		fn block_gas_limit() -> U256 {
+			Revive::evm_block_gas_limit()
+		}
+
+		fn gas_price() -> U256 {
+			Revive::evm_gas_price()
+		}
+
+		fn nonce(address: H160) -> Nonce {
+			let account = <Runtime as pallet_revive::Config>::AddressMapper::to_account_id(&address);
+			System::account_nonce(account)
+		}
+
+		fn eth_transact(tx: pallet_revive::evm::GenericTransaction) -> Result<pallet_revive::EthTransactInfo<Balance>, pallet_revive::EthTransactError>
+		{
+			let blockweights: BlockWeights = <Runtime as frame_system::Config>::BlockWeights::get();
+			let tx_fee = |pallet_call, mut dispatch_info: DispatchInfo| {
+				let call = RuntimeCall::Revive(pallet_call);
+				dispatch_info.extension_weight = EthExtraImpl::get_eth_extension(0, 0u32.into()).weight(&call);
```

Comment:
Why is the nonce and tip not part of the `GenericTransaction` and used here? Not sure if it can make a difference for weight computation though, maybe there is a special path for tip > 0.

Reply:
> Why is the nonce and tip not part of the GenericTransaction

GenericTransaction is an EVM struct, it only defines fields that are available in EVM transaction

> dispatch_info.extension_weight = EthExtraImpl::get_eth_extension(0, 0u32.into()).weight(&call);

we need to set the extension_weight here or the dispatch_weight are not correct, this depends on the tx extensions which get_eth_extension returns, tip and nonce don't really matter here, as they don't impact the weight.